BitwardenShared/UI/Platform/Settings/Settings/AccountSecurity/DeleteAccount/DeleteAccountState.swift

```diff
@@ -2,4 +2,41 @@
 
 /// An object that defines the current state of a `DeleteAccountView`.
 ///
-struct DeleteAccountState: Equatable {}
+struct DeleteAccountState: Equatable {
+    // MARK: Properties
+
+    /// A short description of the view.
+    var description: String {
+        return if shouldPreventUserFromDeletingAccount {
+            Localizations.cannotDeleteAccount
+        } else {
+            Localizations.deletingYourAccountIsPermanent
+        }
+    }
+
+    /// A detailed description of the view.
+    var longDescription: String {
+        return if shouldPreventUserFromDeletingAccount {
+            Localizations.cannotDeleteAccountDescriptionLong
+        } else {
+            Localizations.deleteAccountExplanation
+        }
+    }
+
+    /// The main icon to be displayed.
+    var mainIcon: ImageAsset {
+        return if shouldPreventUserFromDeletingAccount {
+            Asset.Images.circleX16
+        } else {
+            Asset.Images.warning24
+        }
+    }
+
+    /// Whether the form to delete the account is showed.
+    var shouldPreventUserFromDeletingAccount = false
```

Comment:
⛏️ I'd change the docs description because if this is `true` the form is not showed so there isn't an exact correlation with the behavior. Maybe something like:
```suggestion
    /// Whether the user should be blocked from deleting their account.
    var shouldPreventUserFromDeletingAccount = false
```